render/id.go

```diff
@@ -18,10 +18,19 @@ func MakeGroupNodeTopology(originalTopology, key string) string {
 
 // NewDerivedNode makes a node based on node, but with a new ID
 func NewDerivedNode(id string, node report.Node) report.Node {
-	return node.WithID(id).WithChildren(report.MakeNodeSet(node)).PruneParents()
+	return report.MakeNode(id).WithChildren(node.Children.Add(node))
 }
 
 // NewDerivedPseudoNode makes a new pseudo node with the node as a child
 func NewDerivedPseudoNode(id string, node report.Node) report.Node {
-	return NewDerivedNode(id, node).WithTopology(Pseudo)
+	output := NewDerivedNode(id, node).WithTopology(Pseudo)
+
+	// We propagate some metadata here, in a whitelist fashion.
+	if isConnected, timestamp, ok := node.Latest.LookupEntry(IsConnected); ok {
+		output.Latest = output.Latest.Set(IsConnected, timestamp, isConnected)
+	}
+	if hostNodeID, timestamp, ok := node.Latest.LookupEntry(report.HostNodeID); ok {
+		output.Latest = output.Latest.Set(report.HostNodeID, timestamp, hostNodeID)
+	}
```

Comment:
Maybe these should be propagated in `NewDerivedNode`, so they always carry over, not just on pseudonodes? wdyt?